### PR TITLE
[CHORE tests] simplify setupStore registrations

### PR DESCRIPTION
**Pull request by runspired**
This PR allows for more incremental migration away from `createStore` and `setupStore` to modern test syntax.

Specifically it

- ensures that we aren't registering anything unneeded for tests currently
- provides a path for registering a serializer for use with the test
- encourages use of "application" adapter/serializer

cc @pete-the-pete @snewcomer

### CODE REVIEW

**Comment by runspired** [comment]
cc @HeroicEric

File: packages/-ember-data/addon/setup-container.js [hunk 1]
```diff
@@ -31,7 +30,6 @@ function initializeStore(registry) {
   registerOptionsForType.call(registry, 'adapter', { singleton: false });
   registry.register('serializer:-default', JSONSerializer);
   registry.register('serializer:-rest', RESTSerializer);
-  registry.register('adapter:-rest', RESTAdapter);
```
Comment (snewcomer) [inline packages/-ember-data/addon/setup-container.js]: Out of curiosity - this isn't a "breaking" change, right?  (still acclimating)

File: packages/-ember-data/tests/unit/store/adapter-interop-test.js [hunk 2]
```diff
@@ -32,9 +32,12 @@ module('unit/store/adapter-interop - DS.Store working with a DS.Adapter', functi
   });
 
   test('Adapter can be set as a name', function(assert) {
-    store = createStore({ adapter: 'application' });
+    const env = setupStore({ adapter: 'custom-adapter' });
```
Comment (snewcomer) [inline packages/-ember-data/tests/unit/store/adapter-interop-test.js]: I'm sure you already saw this in #6347, but this test was deleted in that PR.

File: packages/-ember-data/tests/helpers/store.js [hunk 3]
```diff
@@ -67,23 +73,31 @@ export default function setupStore(options) {
...
+    // Many tests rely on falling back to this serializer
+    // they should refactor to register this as the application serializer
+    owner.register('serializer:-default', JSONAPISerializer);
+
+    // RESTAdapter specifies a defaultSerializer of -rest
+    // Tests using this should refactor to register this as the application serializer
+    owner.register('serializer:-rest', RESTSerializer);
+
+    env.restSerializer = store.serializerFor('-rest');
+    env.serializer = store.serializerFor('-default');
```
Comment (snewcomer) [inline packages/-ember-data/tests/helpers/store.js]: 👍

File: packages/-ember-data/addon/setup-container.js (hunk 1 above)
Comment (runspired) [reply packages/-ember-data/addon/setup-container.js]: Correct. The existence of this adapter was for tests and private. To use it folks would have had to discover it and looked it up via -rest manually. Given the easy of importing RESTAdapter and no evidence of usage I don’t believe we need to treat this as intimate.

File: packages/-ember-data/tests/unit/store/adapter-interop-test.js (hunk 2 above)
Comment (runspired) [reply packages/-ember-data/tests/unit/store/adapter-interop-test.js]: The removed test was the version of this that set it to an instance, not the one that set it to a string. It does refactor it to not use setupStore/createStore though :)

File: packages/-ember-data/tests/unit/store/adapter-interop-test.js [hunk 4]
```diff
@@ -32,9 +32,12 @@ module('unit/store/adapter-interop - DS.Store working with a DS.Adapter', functi
...
+    const { store, owner } = env;
+    const CustomAdapter = DS.Adapter.extend({ isCustomAdapter: true });
```
Comment (igorT) [inline packages/-ember-data/tests/unit/store/adapter-interop-test.js]: do we need `isCustomAdapter`, makes it look as a special prop

File: packages/-ember-data/tests/unit/store/adapter-interop-test.js (hunk 2 above)
Comment (igorT) [inline packages/-ember-data/tests/unit/store/adapter-interop-test.js]: :+1: on not naming it application, that seemed trolly

File: packages/-ember-data/tests/unit/store/adapter-interop-test.js (hunk 4 above)
Comment (runspired) [reply packages/-ember-data/tests/unit/store/adapter-interop-test.js]: we don't, just trying to make it clear it's custom :D

**Pull request merged**